src/core/api.c

```diff
@@ -1325,20 +1325,32 @@ MsQuicStreamReceiveComplete(
 
     QUIC_CONN_VERIFY(Connection,
         (Stream->RecvPendingLength == 0) || // Stream might have been shutdown already
-        BufferLength <= Stream->RecvPendingLength);
+        BufferLength <= Stream->RecvPendingLength ||
+        BufferLength <= (UINT64_MAX >> 2));
 
     QuicTraceEvent(
         StreamAppReceiveCompleteCall,
         "[strm][%p] Receive complete call [%llu bytes]",
         Stream,
         BufferLength);
 
-    InterlockedExchangeAdd64(
-        (int64_t*)&Stream->RecvCompletionLength, (int64_t)BufferLength);
+    uint64_t RecvCompletionLength = InterlockedExchangeAdd64(
+        (int64_t*)&Stream->RecvCompletionLength,
+        (int64_t)BufferLength);
+    if ((BufferLength & QUIC_STREAM_RECV_COMPLETION_LENGTH_CANARY_BIT) != 0 &&
+        (RecvCompletionLength & QUIC_STREAM_RECV_COMPLETION_LENGTH_CANARY_BIT) != 0) {
+        // overflow detected
```

Comment:
For one, this doesn't follow the comment style of this problem. For another, I think this comment is too trivial. Either we should remove it or expand upon it.